### PR TITLE
xrootd: 5.5.5 -> 5.6.6, python311Packages.xrootd: init at 5.6.6

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Changelog: https://github.com/xrootd/xrootd/releases/tag/v5.6.6

cc @ShamrockLee 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ShamrockLee** [comment]
I tried to package xrootd-python before, but didn't make it build.

**Comment by GaetanLepage** [comment]
> I tried to package xrootd-python before, but didn't make it build.

I was able to make it build.
However I am kindof blocked by this "cycle" issue on the main `xrootd` package...

File: pkgs/development/python-modules/xrootd/default.nix
Comment (SomeoneSerge) [inline pkgs/development/python-modules/xrootd/default.nix:1]: One way to approach this is to use `remove-references-to` and then grep (incl. binaries) for `eeeeeeeeee` after the build

**Comment by GaetanLepage** [comment]
@ofborg build python311Packages.xrootd

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 285505` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li>hepmc3</li>
    <li>ns-3</li>
    <li>python311Packages.hepmc3</li>
    <li>python312Packages.hepmc3</li>
    <li>root</li>
    <li>root.debug</li>
    <li>xfitter</li>
    <li>yoda-with-root</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>python311Packages.xrootd</li>
    <li>python311Packages.xrootd.dist</li>
    <li>python312Packages.xrootd</li>
    <li>python312Packages.xrootd.dist</li>
    <li>xrootd</li>
    <li>xrootd.bin</li>
    <li>xrootd.dev</li>
    <li>xrootd.etc</li>
    <li>xrootd.man</li>
  </ul>
</details>

File: pkgs/development/python-modules/xrootd/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  pname = "xrootd";
+  version = "5.6.6";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "xrootd";
+    repo = "xrootd";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-Su3h1/HLj5bAAnKNKz43C2+DbBUhPWKnqpu80FN6YoA=";
+  };
```
Comment (veprbl) [inline pkgs/development/python-modules/xrootd/default.nix:20]:
```suggestion
  inherit (pkgs.xrootd) version src;
```

Comment (GaetanLepage) [reply pkgs/development/python-modules/xrootd/default.nix:20]: Of course, thanks !
Done.

**Comment by GaetanLepage** [comment]
`root` is not building anymore though:
```
...
-- Found OpenSSL: /nix/store/h24v1xs4i9l3k18ygxvs62361rl77804-openssl-3.0.13/lib/libcrypto.so (found version "3.0.13")  
-- Looking for XROOTD
-- Found Xrootd version num: 5.7-rc19800101 (setting -DROOTXRDVERS=5.7-rc19)
-- Setting OLDPACK TRUE
CMake Error at cmake/modules/SearchInstalledSoftware.cmake:994 (message):
  XROOTD not found.  Set environment variable XRDSYS to point to your XROOTD
  installation, or include the installation of XROOTD in the
  CMAKE_PREFIX_PATH.  Alternatively, you can also enable the option
  'builtin_xrootd' to build XROOTD internally
Call Stack (most recent call first):
  CMakeLists.txt:300 (include)


-- Configuring incomplete, errors occurred!
```

Do we have to adapt [those lines](https://github.com/NixOS/nixpkgs/blob/80e187287de0e1c7307cd3df4708a227425030b1/pkgs/applications/science/misc/root/default.nix#L205-L221) ?

**Comment by SuperSandro2000** [comment]
or it could be that the cmake file is not correct. I am not sure.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 285505` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>17 packages built:</summary>
  <ul>
    <li>hepmc3</li>
    <li>ns-3</li>
    <li>python311Packages.hepmc3</li>
    <li>python311Packages.xrootd</li>
    <li>python311Packages.xrootd.dist</li>
    <li>python312Packages.hepmc3</li>
    <li>python312Packages.xrootd</li>
    <li>python312Packages.xrootd.dist</li>
    <li>root</li>
    <li>root.debug</li>
    <li>xfitter</li>
    <li>xrootd</li>
    <li>xrootd.bin</li>
    <li>xrootd.dev</li>
    <li>xrootd.etc</li>
    <li>xrootd.man</li>
    <li>yoda-with-root</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Thanks so much @veprbl !

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 285505` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages failed to build:</summary>
  <ul>
    <li>hepmc3</li>
    <li>ns-3</li>
    <li>python311Packages.hepmc3</li>
    <li>python312Packages.hepmc3</li>
    <li>root</li>
    <li>xfitter</li>
    <li>yoda-with-root</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>python311Packages.xrootd</li>
    <li>python311Packages.xrootd.dist</li>
    <li>python312Packages.xrootd</li>
    <li>python312Packages.xrootd.dist</li>
    <li>xrootd</li>
    <li>xrootd.bin</li>
    <li>xrootd.dev</li>
    <li>xrootd.etc</li>
    <li>xrootd.man</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
`root` builds fine on darwin on `master`...
Here fails with
```
-- Looking for XROOTD
CMake Error at /nix/store/2p9vvs21zafdidglhrmqn84pjxz9y63v-xrootd-5.6.6-dev/lib/cmake/XRootD/XRootDConfig.cmake:32 (message):
  File or directory referenced by variable XRootD_DATA_DIR does not exist !
Call Stack (most recent call first):
  /nix/store/2p9vvs21zafdidglhrmqn84pjxz9y63v-xrootd-5.6.6-dev/lib/cmake/XRootD/XRootDConfig.cmake:76 (set_and_check)
  cmake/modules/SearchInstalledSoftware.cmake:18 (_find_package)
  cmake/modules/SearchInstalledSoftware.cmake:991 (find_package)
  CMakeLists.txt:300 (include)
```

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 285505` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>ns-3</li>
  </ul>
</details>
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>hepmc3</li>
    <li>python311Packages.hepmc3</li>
    <li>python311Packages.xrootd</li>
    <li>python311Packages.xrootd.dist</li>
    <li>python312Packages.hepmc3</li>
    <li>python312Packages.xrootd</li>
    <li>python312Packages.xrootd.dist</li>
    <li>root</li>
    <li>root.debug</li>
    <li>xfitter</li>
    <li>xrootd</li>
    <li>xrootd.bin</li>
    <li>xrootd.dev</li>
    <li>xrootd.etc</li>
    <li>xrootd.man</li>
    <li>yoda-with-root</li>
  </ul>
</details>

**Comment by ShamrockLee** [comment]
I'll test locally/on LXPLUS these 2 days.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 285505` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 285505` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by GaetanLepage** [comment]
@veprbl Are you sure that you didn't get rid of the darwin packages in your last push ?

**Comment by veprbl** [comment]
> @veprbl Are you sure that you didn't get rid of the darwin packages in your last push ?

Fixed now. Please, try again.

**Comment by GaetanLepage** [comment]
(`aarch64-darwin`)

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 285505` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>ns-3</li>
  </ul>
</details>
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>hepmc3</li>
    <li>python311Packages.hepmc3</li>
    <li>python311Packages.xrootd</li>
    <li>python311Packages.xrootd.dist</li>
    <li>python312Packages.hepmc3</li>
    <li>python312Packages.xrootd</li>
    <li>python312Packages.xrootd.dist</li>
    <li>root</li>
    <li>root.debug</li>
    <li>xfitter</li>
    <li>xrootd</li>
    <li>xrootd.bin</li>
    <li>xrootd.dev</li>
    <li>xrootd.etc</li>
    <li>xrootd.man</li>
    <li>yoda-with-root</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 285505` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>17 packages built:</summary>
  <ul>
    <li>hepmc3</li>
    <li>ns-3</li>
    <li>python311Packages.hepmc3</li>
    <li>python311Packages.xrootd</li>
    <li>python311Packages.xrootd.dist</li>
    <li>python312Packages.hepmc3</li>
    <li>python312Packages.xrootd</li>
    <li>python312Packages.xrootd.dist</li>
    <li>root</li>
    <li>root.debug</li>
    <li>xfitter</li>
    <li>xrootd</li>
    <li>xrootd.bin</li>
    <li>xrootd.dev</li>
    <li>xrootd.etc</li>
    <li>xrootd.man</li>
    <li>yoda-with-root</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 285505` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>ns-3</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>xfitter</li>
  </ul>
</details>
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>hepmc3</li>
    <li>python311Packages.hepmc3</li>
    <li>python311Packages.xrootd</li>
    <li>python311Packages.xrootd.dist</li>
    <li>python312Packages.hepmc3</li>
    <li>python312Packages.xrootd</li>
    <li>python312Packages.xrootd.dist</li>
    <li>root</li>
    <li>scitokens-cpp</li>
    <li>xrootd</li>
    <li>xrootd.bin</li>
    <li>xrootd.dev</li>
    <li>xrootd.etc</li>
    <li>xrootd.man</li>
    <li>yoda-with-root</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
`xfitter-aarch64-darwin` and `xfitter-x86_64-darwin` are also broken on `master`.

**Review by veprbl — APPROVED** [review]
- XRootD.client.file works in python
- xrdcp works
- TFile works in ROOT

**Comment by GaetanLepage** [comment]
Thanks @veprbl !

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 285505` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>xfitter</li>
  </ul>
</details>
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>hepmc3</li>
    <li>ns-3</li>
    <li>python311Packages.hepmc3</li>
    <li>python311Packages.xrootd</li>
    <li>python311Packages.xrootd.dist</li>
    <li>python312Packages.hepmc3</li>
    <li>python312Packages.xrootd</li>
    <li>python312Packages.xrootd.dist</li>
    <li>root</li>
    <li>scitokens-cpp</li>
    <li>xrootd</li>
    <li>xrootd.bin</li>
    <li>xrootd.dev</li>
    <li>xrootd.etc</li>
    <li>xrootd.man</li>
    <li>yoda-with-root</li>
  </ul>
</details>

**Comment by ShamrockLee** [comment]
> I'll test locally/on LXPLUS these 2 days.

I tried PyXRootD today. It largely works with LD_LIBRARY_PATH set to `${xrootd.out}/lib`. However, I haven't got a way to write a Python-based `xrdcp` alternative for `passthru.tests`.

**Comment by ShamrockLee** [comment]
`pkgs` happens to be part of the `<pkg>.override` interface, which is a common anti-pattern that makes it difficult to override the pakage.

The purpose of such `pkgs` in the current expression is to access `pkgs.xrootd` instead of `pythonPackages.xrootd`. In this case, it seems to be a common practice to just name the argument `xrootd`, and specify `{ inherit (pkgs) xrootd; }` for `callPackage`. Just opened #289072 to fix it.

**Comment by veprbl** [comment]
> > I'll test locally/on LXPLUS these 2 days.
> 
> I tried PyXRootD today. It largely works with LD_LIBRARY_PATH set to `${xrootd.out}/lib`. However, I haven't got a way to write a Python-based `xrdcp` alternative for `passthru.tests`.

About LD_LIBRARY_PATH, I did not need it this time. What is a trigger for it? Is there a reproducer with a public URL?

**Comment by veprbl** [comment]
Actually, let's move discussion to #200830

**Pull request merged**